tutorials/shaders/shader_reference/canvas_item_shader.rst

```diff
@@ -155,17 +157,18 @@ texture color by itself, you can use:
 Similarly, if a normal map is used in the :ref:`CanvasTexture <class_CanvasTexture>`, Godot uses
 it by default and assigns its value to the built-in ``NORMAL`` variable. If you are using a normal
 map meant for use in 3D, it will appear inverted. In order to use it in your shader, you must assign
-it to the ``NORMALMAP`` property. Godot will handle converting it for use in 2D and overwriting ``NORMAL``.
+it to the ``NORMAL_MAP`` property. Godot will handle converting it for use in 2D and overwriting ``NORMAL``.
 
 .. code-block:: glsl
 
-  NORMALMAP = texture(NORMAL_TEXTURE, UV).rgb;
+  NORMAL_MAP = texture(NORMAL_TEXTURE, UV).rgb;
 
 +---------------------------------------------+---------------------------------------------------------------+
 | Built-in                                    | Description                                                   |
 +=============================================+===============================================================+
 | in vec4 **FRAGCOORD**                       | Coordinate of pixel center. In screen space. ``xy`` specifies |
-|                                             | position in window. Origin is upper-left.                     |
+|                                             | position in window. Upper-left of the screen is the origin,   |
```

Comment:
position in window or in viewport?

Reply:
I'm not that familiar with 2d coordinate spaces and shaders, I'll investigate this and clarify

Reply:
It seems to be based on the viewport. Here's a quick fragcoord visualization shader, applied on both a full rect control and a subviewport. The subviewport's FRAGCOORD does not change as its position in the window changes.
![Godot_v4 3-stable_win64_Ae2JA0ufjW](https://github.com/user-attachments/assets/7ed7497b-dd95-4e13-aeca-c7afc30caf55)

https://github.com/user-attachments/assets/b3d93a24-b182-4ec6-8e6a-712ecbeef857

Reply:
I figured the original incorrectly stated window instead of viewport.